### ISSUE TITLE
enabling lsp-headerline-breadcrumb causes lsp to stall after a few sessions

### ISSUE THREAD

**Issue by yilkalargaw** [emacs-lsp/lsp-mode#2896]
**Describe the bug**
Enabling lsp-headerline-breadcrumb causes lsp to stall with a message ```Error processing message (void-variable lsp-icons)```.

**To Reproduce**
Use lsp with lsp-headerline-breadcrumb enabled a few times and it will eventually error out with the above message

**Expected behavior**
After enabling lsp-headline-breadcrumb and using lsp a few times it will failing for multiple language server with the the message ```Error processing message (void-variable lsp-icons)``` Disabling this feature will make allow for normal behaviour.

**Which Language Server did you use**
ruby solargraph pyls html-ls

**OS**
Linux ubuntu:21.10

**Error callstack**
```
Debugger entered--Lisp error: (void-variable lsp-icons)
  byte-code("\301\10!\210\302\300!\207" [lsp-icons lsp-consistency-check provide] 2)
  require(lsp-icons)
  byte-code("\300\301!\210\300\302!\210\303\304\305\306\307\310\311\302\312\313&\11\210\314\315\316\317\320DD\321\322\323\311\304&\7\210\314\324\316\317\325DD\326\322\327\311\304&\7..." [require lsp-icons lsp-mode custom-declare-group lsp-headerline nil "LSP support for headerline" :prefix "lsp-headerline-" :group :tag "LSP Headerline" custom-declare-variable lsp-headerline-breadcrumb-segments funcall function #f(compiled-function () #<bytecode 0x84f935cb95b5d>) "Face used on breadcrumb text on modeline." :type (repeat (choice (const :tag "Include the project name." project) (const :tag "Include the open file name." file) (const :tag "Include the directories up to project." path-up-to-project) (const :tag "Include document symbols if server supports it." symbols))) lsp-headerline-breadcrumb-enable-symbol-numbers #f(compiled-function () #<bytecode 0x1f4000153e91>) "Whether to label symbols with numbers on the bread..." boolean lsp-headerline-breadcrumb-enable-diagnostics #f(compiled-function () #<bytecode 0x1f4000153e5d>) "If non-nil, apply different face on the breadcrumb..." :package-version (lsp-mode . "7.1") custom-declare-face lsp-headerline-breadcrumb-separator-face ((t :inherit shadow :height 0.8)) "Face used for breadcrumb separator on headerline." lsp-headerline-breadcrumb-path-face ((t :inherit font-lock-string-face)) "Face used for breadcrumb paths on headerline." lsp-headerline-breadcrumb-path-error-face ((t :underline (:style wave :color "Red1") :inherit lsp-headerline-breadcrumb-path-face)) "Face used for breadcrumb paths on headerline when ..." lsp-headerline-breadcrumb-path-warning-face ((t :underline (:style wave :color "Yellow") :inherit lsp-headerline-breadcrumb-path-face)) "Face used for breadcrumb paths on headerline when ..." lsp-headerline-breadcrumb-path-info-face ((t :underline (:style wave :color "Green") :inherit lsp-headerline-breadcrumb-path-face)) "Face used for breadcrumb paths on headerline when ..." lsp-headerline-breadcrumb-path-hint-face ((t :underline (:style wave :color "Green") :inherit lsp-headerline-breadcrumb-path-face)) "Face used for breadcrumb paths on headerline when ..." lsp-headerline-breadcrumb-project-prefix-face ((t :inherit font-lock-string-face :weight bold)) ...] 10)
  lsp-headerline-breadcrumb-mode()
  lsp-configure-buffer()
  lsp-managed-mode(1)
  lsp--text-document-did-open()
  lsp--open-in-workspace(#s(lsp--workspace :ewoc nil :server-capabilities #<hash-table equal 9/9 0x156d3fafbd61> :registered-server-capabilities nil :root "/home/domatropic/srcs/editors/femto_rb" :client #s(lsp--client :language-id nil :add-on? nil :new-connection (:connect #f(compiled-function (filter sentinel name environment-fn) #<bytecode -0x1f28d91ca8fb7bc4>) :test\? #f(compiled-function () #<bytecode 0x1f6191bf0ef7>)) :ignore-regexps nil :ignore-messages nil :notification-handlers #<hash-table equal 0/65 0x156d3fd56601> :request-handlers #<hash-table equal 0/65 0x156d3fd5682f> :response-handlers #<hash-table eql 1/65 0x156d3fd57f63> :prefix-function nil :uri-handlers #<hash-table equal 0/65 0x156d3fd57f83> :action-handlers #<hash-table equal 0/65 0x156d3fd583c5> :major-modes (ruby-mode enh-ruby-mode) :activation-fn nil :priority -1 :server-id ruby-ls :multi-root t :initialization-options nil :semantic-tokens-faces-overrides nil :custom-capabilities nil :library-folders-fn #f(compiled-function (workspace) #<bytecode -0x47e817621d92e38>) :before-file-open-fn nil :initialized-fn #f(compiled-function (workspace) #<bytecode -0x17787c4d493c9e87>) :remote? nil :completion-in-comments? nil :path->uri-fn nil :uri->path-fn nil :environment-fn nil :after-open-fn nil :async-request-handlers #<hash-table equal 0/65 0x156d3fd587cb> :download-server-fn nil :download-in-progress? nil :buffers nil) :host-root nil :proc #<process ruby-ls> :cmd-proc #<process ruby-ls> :buffers (#<buffer femto.rb>) :semantic-tokens-faces nil :semantic-tokens-modifier-faces nil :extra-client-capabilities nil :status initialized :metadata #<hash-table equal 0/65 0x156d4002089f> :watches #<hash-table equal 0/65 0x156d40020943> :workspace-folders nil :last-id 0 :status-string nil :shutdown-action nil :diagnostics #<hash-table equal 0/65 0x156d4002213d> :work-done-tokens #<hash-table equal 0/65 0x156d400225a9>))
  #f(compiled-function (buffer) #<bytecode -0x7b0d146ff8bf96c>)(#<buffer femto.rb>)
  mapc(#f(compiled-function (buffer) #<bytecode -0x7b0d146ff8bf96c>) (#<buffer femto.rb>))
  #f(compiled-function (response) #<bytecode 0x1c58a2e9bf5d353b>)(#<hash-table equal 1/1 0x156d3fafbd3b>)
  #f(compiled-function (result) #<bytecode -0x95b189eed25d92f>)(#<hash-table equal 1/1 0x156d3fafbd3b>)
  #f(compiled-function (result) #<bytecode -0x7e36465ec7412e8>)(#<hash-table equal 1/1 0x156d3fafbd3b>)
  lsp--parser-on-message(#<hash-table equal 3/3 0x156d3fafbcf5> #s(lsp--workspace :ewoc nil :server-capabilities #<hash-table equal 9/9 0x156d3fafbd61> :registered-server-capabilities nil :root "/home/domatropic/srcs/editors/femto_rb" :client #s(lsp--client :language-id nil :add-on? nil :new-connection (:connect #f(compiled-function (filter sentinel name environment-fn) #<bytecode -0x1f28d91ca8fb7bc4>) :test\? #f(compiled-function () #<bytecode 0x1f6191bf0ef7>)) :ignore-regexps nil :ignore-messages nil :notification-handlers #<hash-table equal 0/65 0x156d3fd56601> :request-handlers #<hash-table equal 0/65 0x156d3fd5682f> :response-handlers #<hash-table eql 1/65 0x156d3fd57f63> :prefix-function nil :uri-handlers #<hash-table equal 0/65 0x156d3fd57f83> :action-handlers #<hash-table equal 0/65 0x156d3fd583c5> :major-modes (ruby-mode enh-ruby-mode) :activation-fn nil :priority -1 :server-id ruby-ls :multi-root t :initialization-options nil :semantic-tokens-faces-overrides nil :custom-capabilities nil :library-folders-fn #f(compiled-function (workspace) #<bytecode -0x47e817621d92e38>) :before-file-open-fn nil :initialized-fn #f(compiled-function (workspace) #<bytecode -0x17787c4d493c9e87>) :remote? nil :completion-in-comments? nil :path->uri-fn nil :uri->path-fn nil :environment-fn nil :after-open-fn nil :async-request-handlers #<hash-table equal 0/65 0x156d3fd587cb> :download-server-fn nil :download-in-progress? nil :buffers nil) :host-root nil :proc #<process ruby-ls> :cmd-proc #<process ruby-ls> :buffers (#<buffer femto.rb>) :semantic-tokens-faces nil :semantic-tokens-modifier-faces nil :extra-client-capabilities nil :status initialized :metadata #<hash-table equal 0/65 0x156d4002089f> :watches #<hash-table equal 0/65 0x156d40020943> :workspace-folders nil :last-id 0 :status-string nil :shutdown-action nil :diagnostics #<hash-table equal 0/65 0x156d4002213d> :work-done-tokens #<hash-table equal 0/65 0x156d400225a9>))
  #f(compiled-function (proc input) #<bytecode -0xd086b9695258229>)(#<process ruby-ls> "Content-Length: 422\15\n\15\n{\"jsonrpc\":\"2.0\",\"id\":1,\"re...")

```

** My LSP setup
``` emacs-lisp
(use-package lsp-mode
  :ensure t
  :defer t
  :init (setq lsp-keymap-prefix "γ l")
  :hook ((ruby-mode . lsp)
         (enh-ruby-mode . lsp)
         (php-mode . lsp)
         (java-mode . lsp)
         (html-mode .lsp)
         (js-mode . lsp)
         ;; (js2-mode . lsp)
         (web-mode . lsp)
         (css-mode . lsp)
         ;; (crystal-mode . lsp)
         (python-mode  . lsp)
         (rust-mode . lsp)
         ;; (go-mode . lsp)
         (c-mode . lsp)
         (c++-mode .lsp)
         ;; (lua-mode . lsp)
         (bash . lsp)
         )
  :config

  ;; change nil to 't to enable logging of packets between emacs and the LS
  ;; this was invaluable for debugging communication with the MS Python Language Server
  ;; and comparing this with what vs.code is doing

  (setq lsp-print-io nil)
  (setq lsp-prefer-flymake nil)
  (setq lsp-enable-file-watchers nil)
  ;; (setq lsp-headerline-breadcrumb-enable nil)
  ;; (setq lsp-headerline-breadcrumb-icons-enable nil)
  ;; (require 'lsp-icons)
  ;; (require 'lsp-clients)

  )


;; lsp-ui gives us the blue documentation boxes and the sidebar info
(use-package lsp-ui
  :ensure t
  :hook (lsp-mode . lsp-ui-mode)
  :config
  (setq lsp-ui-doc-enable t
        lsp-ui-doc-use-childframe t
        lsp-ui-doc-position 'top
        lsp-ui-doc-include-signature t
        lsp-ui-sideline-enable t
        lsp-ui-flycheck-enable t
        lsp-ui-flycheck-list-position 'right
        lsp-ui-flycheck-live-reporting t
        lsp-ui-peek-enable t
        lsp-ui-peek-list-width 60
        lsp-ui-peek-peek-height 25)
  (setq lsp-ui-sideline-ignore-duplicate t)
  ;; (add-hook 'lsp-mode-hook 'lsp-ui-mode)

  (define-key lsp-ui-mode-map [remap xref-find-definitions] #'lsp-ui-peek-find-definitions)
  (define-key lsp-ui-mode-map [remap xref-find-references] #'lsp-ui-peek-find-references)
  )

;;(use-package company-lsp :ensure t :defer t :hook (lsp-mode))

;; (setq lsp-language-id-configuration '(;; (java-mode . "java")
;;                                    ;; (python-mode . "python")
;;                                    ;; (gfm-view-mode . "markdown")
;;                                    ;; (rust-mode . "rust")
;;                                    (ruby-mode . "ruby")
;;                                    ;; (css-mode . "css")
;;                                    ;; (xml-mode . "xml")
;;                                    ;; (c-mode . "c")
;;                                    ;; (c++-mode . "cpp")
;;                                    ;; (objc-mode . "objective-c")
;;                                    ;; (web-mode . "html")
;;                                    ;; (html-mode . "html")
;;                                    ;; (sgml-mode . "html")
;;                                    ;; (mhtml-mode . "html")
;;                                    ;; (go-mode . "go")
;;                                    ;; (haskell-mode . "haskell")
;;                                    ;; (php-mode . "php")
;;                                    ;; (json-mode . "json")
;;                                    ;; (js2-mode . "javascript")
;;                                    ;;(typescript-mode . "typescript")

;;                                    ))

;; LSP debugging

;;(setq lsp-print-io t)

;;(setq lsp-trace t)

;;(setq lsp-print-performance t)

(use-package dap-mode
  :ensure t ;;:after lsp-mode
  :defer t
  :hook (lsp-mode . dap-mode)
  :config
  ;;(dap-mode t)
  (dap-ui-mode t)
  (require 'dap-ruby)
  (require 'dap-php)
  (require 'dap-java)
  )

(use-package lsp-ivy :ensure t :defer t :after lsp)
(use-package lsp-treemacs :ensure t :defer t :after lsp)

```

**Comment by yyoncho** [comment]
It looks like your installation is corrupted. Check if lsp-icons.el is present in lsp-mode folder, you may use `find-library` as well.

**Comment by cbsirb** [comment]
Adding a `(require 'lsp-mode)` in `lsp-icons.el` fixes this issue. There are some errors and warnings related to this during native compilation without the mentioned require:
```
lsp-icons.el:92:2: Warning: the function ‘lsp-consistency-check’ is not known to be defined.
Compiling emacs.d/elpa/lsp-mode-20210525.2051/lsp-headerline.el...
emacs.d/elpa/lsp-mode-20210525.2051/lsp-headerline.el: Error: Symbol's value as variable is void lsp-icons
```

**Comment by thermopyle** [comment]
I experience this as well. I updated a couple of days ago, since then I get this error message.

**Comment by yyoncho** [comment]
Fixed with #2897
